ballerina/pdDocument.bal

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# The class that coresponds to org.apache.pdfbox.pdmodel.PDDocument.
+# This is the in-memory representation of a PDF document.
+public distinct class PDDocument {
+    private handle jObj;
```

Comment:
Instead of maintaining a variable for `handle` instance, we can store the object as a native data from the java side?

Reply:
Here, the PDDocument corresponds to the `org.apache.pdfbox.pdmodel.PDDocument` class coming from the `apache pdfbox` library. Therefore, to avoid having a handle here and to move it to Java side, we need to create a java wrapper for each `pdfbox` class.

Furthermore, if the purpose of not having a `handle` instance in the Ballerina `PDDocument` class is to avoid letting user pass a handle to the` init` method, in this case, I am accepting only ballerina types in the `init` method. Hence, this will not be an issue.

WDYT?

Reply:
@gayaldassanayake convention is to write a java-wrapper whenever we need to access Java objects/clients to perform a particular actions. And we usually hide the relevant Java object/client instance as the native data. IMO adding fields with non-ballerina types (like `handle`) inside Ballerina classes is an anti-pattern.